action.yml

```
@@ -25,15 +25,30 @@ inputs:
 runs:
   using: "composite"
   steps:
-    - uses: actions/setup-java@v4
+    - name: Setup Java 11
+      uses: actions/setup-java@v4
+      if: ${{ inputs.daily == 'false' || startsWith( inputs.version, '3.' ) || startsWith( inputs.version, '4.' ) }}
       with:
         java-version: 11
         distribution: zulu
 
+    - name: Setup Java 17
+      uses: actions/setup-java@v4
+      if: ${{ inputs.daily == 'true' || (!startsWith( inputs.version, '3.' ) && !startsWith( inputs.version, '4.' )) }}
+      with:
+        java-version: 17
+        distribution: zulu
```

Should we simplify this bit by always installing both?

```suggestion
    - name: Setup Java
      uses: actions/setup-java@v4
      with:
        java-version: |
          11
          17
        distribution: zulu
```

We'd keep the logic on the lower bits that actually sets JAVA_HOME. Unsure on complexity vs execution time trade-off 🤔

---

Is there any way that you can think of that we can first calculate the desired value (`11` or `17`) and then just use that as a variable?

---

I was just looking at that in an IDE. Absolutely.

---

Pushed a change.